### PR TITLE
mpir: patch for Visual Studio 2019 and 2022

### PR DESCRIPTION
**Pull request by fpelliccioni**
Specify library name and version:  **MPIR/3.0.0**

MPIR recipe does not work if you have just Visual Studio 2022 or 2019 installed.

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by MartinDelille** [comment]
Can you add the recipe's name in the PR?

**Comment by fpelliccioni** [comment]
> Can you add the recipe's name in the PR?

done

**Comment by uilianries** [comment]
@fpelliccioni Please, consider https://github.com/fpelliccioni/conan-center-index/pull/1

**Comment by fpelliccioni** [comment]
> @fpelliccioni Please, consider [fpelliccioni#1](https://github.com/fpelliccioni/conan-center-index/pull/1)

Thanks @uilianries 
Merged

File: recipes/mpir/all/conanfile.py [hunk 1]
```diff
@@ -110,7 +109,8 @@ def _build_visual_studio(self):
                 "Debug" if "d" in msvc_runtime_flag(self) else "",
                 "DLL" if "MD" in msvc_runtime_flag(self) else "",
             )
-            tools.replace_in_file(props_path, old_runtime, new_runtime)
+            self.output.info(f"--++--++--++--++--++--++--++--++--++--++ props_path: {props_path}")
```
Comment (fpelliccioni) [inline recipes/mpir/all/conanfile.py:112]:
```suggestion
```

**Comment by fpelliccioni** [comment]
> Failure in build 20 (`35533ca48800e417e79c8803401247f19ad92218`):
> 
> * **mpir/3.0.0@:**
>   CI failed to create some packages ([All logs](https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/12204/20-linux-gcc/mpir/3.0.0//summary.json))
>   Logs for packageID `1f170d44f1cc732dc1c1d5936470f2f4d4950332`:
>   ```
>   [settings]
>   arch=x86_64
>   build_type=Release
>   compiler=gcc
>   compiler.libcxx=libstdc++
>   compiler.version=10
>   os=Linux
>   [options]
>   mpir:shared=False
>   ```
>   
>   
>       
>         
>       
>   
>         
>       
>   
>       
>     
>   ```
>   [...]
>   libtool: install: chmod 644 /home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/package/1f170d44f1cc732dc1c1d5936470f2f4d4950332/lib/libmpirxx.a
>   libtool: install: ranlib /home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/package/1f170d44f1cc732dc1c1d5936470f2f4d4950332/lib/libmpirxx.a
>   libtool: install: /usr/bin/install -c .libs/libgmpxx.a /home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/package/1f170d44f1cc732dc1c1d5936470f2f4d4950332/lib/libgmpxx.a
>   libtool: install: chmod 644 /home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/package/1f170d44f1cc732dc1c1d5936470f2f4d4950332/lib/libgmpxx.a
>   libtool: install: ranlib /home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/package/1f170d44f1cc732dc1c1d5936470f2f4d4950332/lib/libgmpxx.a
>   libtool: finish: PATH="/home/conan/w/prod/BuildSingleReference/.conan/data/yasm/1.3.0/_/_/package/24647d9fe8ec489125dfbae4b3ebefaf7581674c/bin:/home/conan/w/prod/BuildSingleReference/.conan/data/m4/1.4.19/_/_/package/24647d9fe8ec489125dfbae4b3ebefaf7581674c/bin:/opt/pyenv/versions/3.7.13/bin:/opt/pyenv/libexec:/opt/pyenv/plugins/python-build/bin:/opt/pyenv/plugins/pyenv-virtualenv/bin:/opt/pyenv/plugins/pyenv-update/bin:/opt/pyenv/plugins/pyenv-installer/bin:/opt/pyenv/plugins/pyenv-doctor/bin:/opt/pyenv/shims:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/sbin" ldconfig -n /home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/package/1f170d44f1cc732dc1c1d5936470f2f4d4950332/lib
>   ----------------------------------------------------------------------
>   Libraries have been installed in:
>      /home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/package/1f170d44f1cc732dc1c1d5936470f2f4d4950332/lib
>   
>   If you ever happen to want to link against installed libraries
>   in a given directory, LIBDIR, you must either use libtool, and
>   specify the full pathname of the library, or use the `-LLIBDIR'
>   flag during linking and do at least one of the following:
>      - add LIBDIR to the `LD_LIBRARY_PATH' environment variable
>        during execution
>      - add LIBDIR to the `LD_RUN_PATH' environment variable
>        during linking
>      - use the `-Wl,-rpath -Wl,LIBDIR' linker flag
>      - have your system administrator add LIBDIR to `/etc/ld.so.conf'
>   
>   See any operating system documentation about shared libraries for
>   more information, such as the ld(1) and ld.so(8) manual pages.
>   ----------------------------------------------------------------------
>   make[3]: Leaving directory '/home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/build/1f170d44f1cc732dc1c1d5936470f2f4d4950332/source_subfolder'
>   make[2]: Leaving directory '/home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/build/1f170d44f1cc732dc1c1d5936470f2f4d4950332/source_subfolder'
>   make[1]: Leaving directory '/home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/build/1f170d44f1cc732dc1c1d5936470f2f4d4950332/source_subfolder'
>   [HOOK - conan-center.py] post_package(): [DEFAULT PACKAGE LAYOUT (KB-H013)] OK
>   [HOOK - conan-center.py] post_package(): [MATCHING CONFIGURATION (KB-H014)] OK
>   [HOOK - conan-center.py] post_package(): [SHARED ARTIFACTS (KB-H015)] OK
>   [HOOK - conan-center.py] post_package(): [PC-FILES (KB-H020)] OK
>   [HOOK - conan-center.py] post_package(): [CMAKE-MODULES-CONFIG-FILES (KB-H016)] OK
>   [HOOK - conan-center.py] post_package(): [PDB FILES NOT ALLOWED (KB-H017)] OK
>   [HOOK - conan-center.py] post_package(): [LIBTOOL FILES PRESENCE (KB-H018)] OK
>   [HOOK - conan-center.py] post_package(): [MS RUNTIME FILES (KB-H021)] OK
>   [HOOK - conan-center.py] post_package(): [SHORT_PATHS USAGE (KB-H066)] OK
>   [HOOK - conan-center.py] post_package(): [MISSING SYSTEM LIBS (KB-H043)] OK
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   [HOOK - conan-center.py] post_package(): ERROR: [PACKAGE LICENSE (KB-H012)] No 'licenses' folder found in package: /home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/package/1f170d44f1cc732dc1c1d5936470f2f4d4950332  (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H012) 
>   ERROR: [HOOK - conan-center.py] post_package(): Some checks failed running the hook, check the output
>   ```
> 
> **Note:** To save resources, CI tries to finish as soon as an error is found. For this reason you might find that not all the references have been launched or not all the configurations for a given reference. Also, take into account that we cannot guarantee the order of execution as it depends on CI workload and workers availability.



> Failure in build 20 (`35533ca48800e417e79c8803401247f19ad92218`):
> 
> * **mpir/3.0.0@:**
>   CI failed to create some packages ([All logs](https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/12204/20-linux-gcc/mpir/3.0.0//summary.json))
>   Logs for packageID `1f170d44f1cc732dc1c1d5936470f2f4d4950332`:
>   ```
>   [settings]
>   arch=x86_64
>   build_type=Release
>   compiler=gcc
>   compiler.libcxx=libstdc++
>   compiler.version=10
>   os=Linux
>   [options]
>   mpir:shared=False
>   ```
>   
>   
>       
>         
>       
>   
>         
>       
>   
>       
>     
>   ```
>   [...]
>   libtool: install: chmod 644 /home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/package/1f170d44f1cc732dc1c1d5936470f2f4d4950332/lib/libmpirxx.a
>   libtool: install: ranlib /home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/package/1f170d44f1cc732dc1c1d5936470f2f4d4950332/lib/libmpirxx.a
>   libtool: install: /usr/bin/install -c .libs/libgmpxx.a /home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/package/1f170d44f1cc732dc1c1d5936470f2f4d4950332/lib/libgmpxx.a
>   libtool: install: chmod 644 /home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/package/1f170d44f1cc732dc1c1d5936470f2f4d4950332/lib/libgmpxx.a
>   libtool: install: ranlib /home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/package/1f170d44f1cc732dc1c1d5936470f2f4d4950332/lib/libgmpxx.a
>   libtool: finish: PATH="/home/conan/w/prod/BuildSingleReference/.conan/data/yasm/1.3.0/_/_/package/24647d9fe8ec489125dfbae4b3ebefaf7581674c/bin:/home/conan/w/prod/BuildSingleReference/.conan/data/m4/1.4.19/_/_/package/24647d9fe8ec489125dfbae4b3ebefaf7581674c/bin:/opt/pyenv/versions/3.7.13/bin:/opt/pyenv/libexec:/opt/pyenv/plugins/python-build/bin:/opt/pyenv/plugins/pyenv-virtualenv/bin:/opt/pyenv/plugins/pyenv-update/bin:/opt/pyenv/plugins/pyenv-installer/bin:/opt/pyenv/plugins/pyenv-doctor/bin:/opt/pyenv/shims:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/sbin" ldconfig -n /home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/package/1f170d44f1cc732dc1c1d5936470f2f4d4950332/lib
>   ----------------------------------------------------------------------
>   Libraries have been installed in:
>      /home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/package/1f170d44f1cc732dc1c1d5936470f2f4d4950332/lib
>   
>   If you ever happen to want to link against installed libraries
>   in a given directory, LIBDIR, you must either use libtool, and
>   specify the full pathname of the library, or use the `-LLIBDIR'
>   flag during linking and do at least one of the following:
>      - add LIBDIR to the `LD_LIBRARY_PATH' environment variable
>        during execution
>      - add LIBDIR to the `LD_RUN_PATH' environment variable
>        during linking
>      - use the `-Wl,-rpath -Wl,LIBDIR' linker flag
>      - have your system administrator add LIBDIR to `/etc/ld.so.conf'
>   
>   See any operating system documentation about shared libraries for
>   more information, such as the ld(1) and ld.so(8) manual pages.
>   ----------------------------------------------------------------------
>   make[3]: Leaving directory '/home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/build/1f170d44f1cc732dc1c1d5936470f2f4d4950332/source_subfolder'
>   make[2]: Leaving directory '/home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/build/1f170d44f1cc732dc1c1d5936470f2f4d4950332/source_subfolder'
>   make[1]: Leaving directory '/home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/build/1f170d44f1cc732dc1c1d5936470f2f4d4950332/source_subfolder'
>   [HOOK - conan-center.py] post_package(): [DEFAULT PACKAGE LAYOUT (KB-H013)] OK
>   [HOOK - conan-center.py] post_package(): [MATCHING CONFIGURATION (KB-H014)] OK
>   [HOOK - conan-center.py] post_package(): [SHARED ARTIFACTS (KB-H015)] OK
>   [HOOK - conan-center.py] post_package(): [PC-FILES (KB-H020)] OK
>   [HOOK - conan-center.py] post_package(): [CMAKE-MODULES-CONFIG-FILES (KB-H016)] OK
>   [HOOK - conan-center.py] post_package(): [PDB FILES NOT ALLOWED (KB-H017)] OK
>   [HOOK - conan-center.py] post_package(): [LIBTOOL FILES PRESENCE (KB-H018)] OK
>   [HOOK - conan-center.py] post_package(): [MS RUNTIME FILES (KB-H021)] OK
>   [HOOK - conan-center.py] post_package(): [SHORT_PATHS USAGE (KB-H066)] OK
>   [HOOK - conan-center.py] post_package(): [MISSING SYSTEM LIBS (KB-H043)] OK
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   cc1plus: warning: command-line option '-Wno-implicit-function-declaration' is valid for C/ObjC but not for C++
>   [HOOK - conan-center.py] post_package(): ERROR: [PACKAGE LICENSE (KB-H012)] No 'licenses' folder found in package: /home/conan/w/prod/BuildSingleReference/.conan/data/mpir/3.0.0/_/_/package/1f170d44f1cc732dc1c1d5936470f2f4d4950332  (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H012) 
>   ERROR: [HOOK - conan-center.py] post_package(): Some checks failed running the hook, check the output
>   ```
> 
> **Note:** To save resources, CI tries to finish as soon as an error is found. For this reason you might find that not all the references have been launched or not all the configurations for a given reference. Also, take into account that we cannot guarantee the order of execution as it depends on CI workload and workers availability.

@uilianries now the CCI is failing after I merged your PR.
I do not know why

File: recipes/mpir/all/conanfile.py [hunk 2]
```diff
@@ -150,43 +149,67 @@ def _configure_autotools(self):
...
-                tools.replace_in_file("configure", "-install_name \\$rpath/", "-install_name @rpath/")
+                replace_in_file(self, "configure", "-install_name \\$rpath/", "-install_name @rpath/")
                 autotools = self._configure_autotools()
                 autotools.make()
 
     def package(self):
-        self.copy("COPYING*", dst="licenses", src=self._source_subfolder)
-        if self._is_msvc:
-            lib_folder = os.path.join(self._source_subfolder, self._dll_or_lib,
+        copy(self, "COPYING*", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
```
Comment (uilianries) [inline recipes/mpir/all/conanfile.py:186]:
```suggestion
        copy(self, "COPYING*", dst=os.path.join(self.package_folder, "licenses"), src=os.path.join(self.source_folder, self._source_subfolder))
```

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**